File: backend-express/src/scripts/testStorage.ts
import { storage } from "@/lib/clients";
import { closeAll, startAll } from "@/lib/setup";
import { getImagePath } from "@/static";

async function test() {
    const path = getImagePath("avatar1.jpg");
    const destination = await storage.uploadFile(path);
    const url = await storage.getSignedUrl(destination);
    console.log(url);
}

if (require.main === module) {
    startAll().then(test).finally(closeAll);
}

File: backend-gin/internal/scripts/test_storage.go
package scripts

import (
	"backend/internal/lib/clients"
	"backend/internal/lib/setup"
	"backend/internal/static"
	"fmt"
)

func TestStorage() {
	setup := setup.New()
	defer setup.CloseSerives()

	path := static.GetImagePath("avatar1.jpg")
	storage := clients.GetStorage()
	defer storage.Close()
	destination, err := storage.UploadFile(path)
	if err != nil {
		fmt.Println(err.Error())
	}

	url, err := storage.GetSignedUrl(destination)
	if err != nil {
		fmt.Println(err)
	} else {
		fmt.Println(url)
	}
}
